test/manipulation.test.js

```diff
@@ -261,6 +435,30 @@ describe('manipulation', function () {
       });
     });
 
+    it('should save invalid object (skipping validation - promise variant)', function (done) {
+      Person.findOne()
+      .then(function (p) {
+        p.isValid = function (done) {
+          process.nextTick(done);
+          return false;
+        };
+        p.name = 'Nana';
+        return p.save()
+        .then(function (d) {
+          should.not.exist(d);
+          done();
+        })
+        .catch(function (err) {
+          should.exist(err);
+          p.save({validate: false})
+          .then(function (d) {
+            should.exist(d);
+            done();
+          }).catch(done);
+        });
+      });
+    });
```

Comment:
The same problem as described in my [comment above](#r25005283), if the operation is supposed to fail, then the `.then()` block should assert on that.